src/tests/Builder.test.js

```diff
@@ -0,0 +1,594 @@
+         expect(builder.isExcludedResource(tbl2W)).to.be(false);
+         expect(builder.isExcludedResource(tbl2idR)).to.be(false);
+         expect(builder.isExcludedResource(tbl2idW)).to.be(false);
+
+         builder.excludeTable(tbl1W.tableName, tbl1W.capacityType);
+         expect(builder.isExcludedResource(tbl1R)).to.be(true);
+         expect(builder.isExcludedResource(tbl1W)).to.be(true);
+         expect(builder.isExcludedResource(tbl1idR)).to.be(false);
+         expect(builder.isExcludedResource(tbl1idW)).to.be(false);
+         expect(builder.isExcludedResource(tbl2R)).to.be(false);
+         expect(builder.isExcludedResource(tbl2W)).to.be(false);
+         expect(builder.isExcludedResource(tbl2idR)).to.be(false);
+         expect(builder.isExcludedResource(tbl2idW)).to.be(false);
+      });
+
+      it('excludes a particular resource for capacity type - indexes', function() {
+         builder.excludeIndex(tbl1idR.tableName, tbl1idR.indexName, tbl1idR.capacityType);
+         expect(builder.isExcludedResource(tbl1R)).to.be(false);
+         expect(builder.isExcludedResource(tbl1W)).to.be(false);
+         expect(builder.isExcludedResource(tbl1idR)).to.be(true);
+         expect(builder.isExcludedResource(tbl1idW)).to.be(false);
+         expect(builder.isExcludedResource(tbl2R)).to.be(false);
+         expect(builder.isExcludedResource(tbl2W)).to.be(false);
+         expect(builder.isExcludedResource(tbl2idR)).to.be(false);
+         expect(builder.isExcludedResource(tbl2idW)).to.be(false);
+
+         builder.excludeIndex(tbl1idW.tableName, tbl1idW.indexName, tbl1idW.capacityType);
+         expect(builder.isExcludedResource(tbl1R)).to.be(false);
+         expect(builder.isExcludedResource(tbl1W)).to.be(false);
+         expect(builder.isExcludedResource(tbl1idR)).to.be(true);
+         expect(builder.isExcludedResource(tbl1idW)).to.be(true);
+         expect(builder.isExcludedResource(tbl2R)).to.be(false);
+         expect(builder.isExcludedResource(tbl2W)).to.be(false);
+         expect(builder.isExcludedResource(tbl2idR)).to.be(false);
+         expect(builder.isExcludedResource(tbl2idW)).to.be(false);
+      });
+
+      it('excludes multiple resource for capacity type with wildcard - tables', function() {
+         builder.excludeTable('Tbl1*', constants.READ);
+         expect(builder.isExcludedResource(tbl1R)).to.be(true);
+         expect(builder.isExcludedResource(tbl1W)).to.be(false);
+         expect(builder.isExcludedResource(tbl1idR)).to.be(false);
+         expect(builder.isExcludedResource(tbl1idW)).to.be(false);
+         expect(builder.isExcludedResource(tbl2R)).to.be(false);
+         expect(builder.isExcludedResource(tbl2W)).to.be(false);
+         expect(builder.isExcludedResource(tbl2idR)).to.be(false);
+         expect(builder.isExcludedResource(tbl2idW)).to.be(false);
+
+         builder.excludeTable('Tbl*', constants.WRITE);
+         expect(builder.isExcludedResource(tbl1R)).to.be(true);
```

Comment:
At first I thought "how is this passing". Then I realized that it's passing because it's re-using the state from above. Okay, that makes sense - and may move you more towards just changing the `it` description. Still, I think a comment here (and places like it) would be good ... maybe just an EOL comment `// excluded by previous builder call` or something.